src/components/ModalFlows/SupportProposal/SupportProposalScreens.js

```diff
@@ -1,39 +1,29 @@
 import React, { useMemo, useState, useCallback } from 'react'
-import PropTypes from 'prop-types'
 import ModalFlowBase from '../ModalFlowBase'
 import SupportProposal from './SupportProposal'
-import useActions from '../../../hooks/useActions'
 
-function SupportProposalScreens({ versionId, id, onDone, stakeProposal }) {
-  const actions = useActions()
+function SupportProposalScreens({ onStakeToProposal, id }) {
   const [transactions, setTransactions] = useState([])
 
   const getTransactions = useCallback(
-    async onComplete => {
-      await actions.agreementActions.signAgreement({ versionId }, intent => {
+    async (onComplete, proposalId, amount) => {
+      await onStakeToProposal({ proposalId, amount }, intent => {
         setTransactions(intent.transactions)
         onComplete()
       })
     },
-    [actions, versionId]
+    [onStakeToProposal]
   )
 
   const screens = useMemo(
     () => [
       {
         title: 'Support this proposal',
         graphicHeader: false,
-        content: (
-          <SupportProposal
-            id={id}
-            getTransactions={getTransactions}
-            onDone={onDone}
-            onStakeProposal={stakeProposal}
-          />
-        ),
+        content: <SupportProposal id={id} getTransactions={getTransactions} />,
```

Comment:
Not sure if you need to pass the id to this component since you already have it in this component at the moment of calling the `stakeToProposal` here https://github.com/1Hive/honey-pot/pull/206/files/1dd37543ec0308d33c78da22fe5251c2d86386c2..faa8b3902b58242e2e79ceb730d6ca4b4593e202#diff-5ecbb913608e7df708a2dc6c7f21e7e034675b15e4a9ba11aa8f277cbcce968bR10 , is the id that you have in the propoerties, so here you are passing it to the `<SupportProposal>` component to pass it back to the parent here https://github.com/1Hive/honey-pot/pull/206/files/1dd37543ec0308d33c78da22fe5251c2d86386c2..faa8b3902b58242e2e79ceb730d6ca4b4593e202#diff-5daa6f474d614ca00802abd12c3e990e03506232d87e08dd10a30f0964c2c7dfR48